lemur/plugins/lemur_acme/plugin.py

```diff
@@ -91,6 +92,7 @@ def start_dns_challenge(
         self,
         acme_client,
         account_number,
+        domain,
```

Comment:
I wonder if we could go with a better naming for `domain` and `host`, since they are overloaded here.
maybe `cnamed_host`, to indicate the purpose of host, which should help readability.

Reply:
These names reflect what ultimately ends up in the AuthorizationRecord. If renamed, it should probably be consistent.  The challenge is that the host is used regardless of whether there is a CNAME or not, so I don't think cname_host is ideal.  Perhaps something like target_domain.

Reply:
naming is hard. let's go with `target_name` for now